x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -65,6 +65,18 @@ export type EventCategoryOverride = t.TypeOf<typeof event_category_override>;
 export const eventCategoryOverrideOrUndefined = t.union([event_category_override, t.undefined]);
 export type EventCategoryOverrideOrUndefined = t.TypeOf<typeof eventCategoryOverrideOrUndefined>;
 
+export const tiebreaker_field = t.string;
+export type TiebreakerField = t.TypeOf<typeof tiebreaker_field>;
+
+export const tiebreakerFieldOrUndefined = t.union([tiebreaker_field, t.undefined]);
+export type TiebreakerFieldOrUndefined = t.TypeOf<typeof tiebreakerFieldOrUndefined>;
+
+export const timestamp_field = t.string;
+export type TimestampField = t.TypeOf<typeof timestamp_field>;
+
+export const timestampFieldOrUndefined = t.union([timestamp_field, t.undefined]);
+export type TimestampFieldOrUndefined = t.TypeOf<typeof timestampFieldOrUndefined>;
```

Comment:
So we already have `timestamp_override` which I guess has the same semantics? We're just adding support for `timestamp_override` for EQL rules, right?

It looks confusing to see both `timestamp_override` and `timestamp_field` in the rule schemas like for example `addPrepackagedRulesSchema`, adds some inconsistency and forces rule authors to remember which one to use depending on the rule type.

I'd suggest to remove the new `timestamp_field` from all schemas in favor of reusing the existing `timestamp_override` field.

Reply:
I had exactly the same confusion about these two variables, though it seems like we use `timestamp_override` in a different place (`Create Rule > 2. About Rule > Advanced settings > Timestamp override`) for different purposes. @marshallmain could you add more information about this one?

Reply:
EQL rules already support `timestamp_override` with the same semantics as other rule types. The new `timestamp_field` is a pass through to the [EQL API's ](https://www.elastic.co/guide/en/elasticsearch/reference/current/eql-search-api.html#eql-search-api-request-body). At a high level, `timestamp_override` controls whether or not a document is in range of the EQL query *at all*, whereas the `timestamp_field` controls how documents are sorted within the set of documents that are in range. Typically we want to use `event.ingested` for `timestamp_override` to ensure that the detection engine searches every document as they come in. For `timestamp_field` though users may want to use `@timestamp` or another field that represents the time the event happened *on the source* so that events are ordered correctly for sequences.

Reply:
Oh, this is really interesting, gotcha, thanks for the explanation @marshallmain! Can we put it into a jsdoc comment for the `timestamp_field` schema above?

A few quadruple-check questions:

- Can an EQL rule have both `timestamp_override` and `timestamp_field` fields initialized with different values? Like you said, `timestamp_override` can be `event.ingested` and `timestamp_field` can be `@timestamp`.
- Can an EQL rule have both fields initialized to the same value?
- Can an EQL rule have only one of the fields, e.g. `timestamp_field` but not `timestamp_override`?
- If not specified, does `timestamp_field` default to some value? Does it default to `timestamp_override` if the latter is specified?

Reply:
@marshallmain correct me if I'm wrong, but here is my understanding:

- Can an EQL rule have both timestamp_override and timestamp_field fields initialized with different values? Like you said, timestamp_override can be event.ingested and timestamp_field can be @timestamp.

> Yes. Both are independent and can be initialised with different values.

- Can an EQL rule have both fields initialized to the same value?

> Yes. We will use the same field to query documents and then to sort documents in range.

- Can an EQL rule have only one of the fields, e.g. timestamp_field but not timestamp_override?

> Yes. Both default to `@timestamp`.

- If not specified, does timestamp_field default to some value? Does it default to timestamp_override if the latter is specified?

> Both default to `@timestamp`.